docs/sections/user_guide/rocoto_yaml.rst

```diff
@@ -0,0 +1,606 @@
+
+The name of the task can be any string accepted by Rocoto as a task name (including additional underscores), but must contain the leading ``task_`` to be recognized as a task.
+
+``command``: The command that will be run in the batch job.
+
+``envars``: Any number of key/value pairs to set up the environment for the ``<command>`` to run successfully. Here, keys translate to bash variable names and values translate to the bash variables' values. Each entry in this section will show up in the XML like this:
+
+.. code:: XML
+
+  <envar>
+    <name>person</name>
+    <value>siri</value>
+  </envar>
+
+``dependencies``: [Optional] Any number of dependencies accepted by Rocoto. This entry is described in more detail below.
+
+
+The other tags not specifically mentioned here are follow the same conventions as described in the :rocoto:`Rocoto<>` documentation.
+
+
+
+Defining Dependencies for Tasks
+...............................
+
+Rocoto dependencies are optional entries that are structured as boolean expressions defining the readiness of a task to be submitted to the queue. The :rocoto:`Rocoto documentation<>` explains each tag in detail. Here is an explanation for how those tags should be specified in YAML format.
+
+There are many similarities, but some nuanced differences must be clarified.
+
+Each dependency represented in the YAML will be named exactly the same as a dependency in Rocoto XML, but can be suffixed with an arbitrary descriptor for the dependency after an underscore (``_``). For example, a ``<streq>`` tag could be in the YAML under ``streq_check_flag:`` or similar. Additional underscores are allowed.
+
+Tag Attributes
+______________
+
+Each of the dependencies that require XML attributes (the "key=value" parts inside the XML tag) can be specified with an ``attrs`` entry. For example:
+
+.. code::
+
+  task_hello:
+    command: "hello world"
+    ...
+  task_goodbye:
+    command: "goodbye"
+    dependencies:
+       taskdep:
+         attrs:
+           task: hello
+
+Here, the Rocoto ``taskdep`` says that the ``goodbye`` task cannot be submitted until the ``hello`` task is complete. This will result in Rocoto XML that looks like the following snippet:
+
+.. code:: XML
+
+  <task name="hello">
+    ...
+  </task>
+  <task name="goodbye"/>
+    ...
+    <dependency>
+      <taskdep task="hello"/>
+    </dependency>
+  </task>
+
+Repeated Dependencies and Boolean Operators
+___________________________________________
+
+Because YAML represents a hash table (a dictionary in Python), each entry at the same level must be unique. To accomplish this in the YAML format, any of the dependencies may be specified with an arbitrary unique suffix following an underscore (``_``). We recommend a descriptive one to make it easier to read. In the following example, there are multiple data dependencies for the basic ``hello`` task.
```

Comment:
I understand where you're coming from. It's not so much about the order here, but the ability for a user to override or remove one of these settings using our standard "update_values" method. I have made a note to add a ticket for us to address this difference in behavior for the UW config tool and the SRW version...

Users may want to add a new dependency to existing ones, and can do that by adding a supplemental file to the `uw config realize` step. They may also want to remove an existing one, which should be accomplished by setting an entry to empty (this feature is what the ticket will call for). I don't see a straightforward way to modify a base workflow definition that is as straightforward when we use a sequence rather than a mapping.

Reply:
Updating a specific task entry is a good point / use case, and this may not be the first time you've mentioned it. I'll try to remember that that's the goal -- and maybe _not_ say this again. Provided that reordering would be acceptable if that's what a YAML or map/hash/dict library decided to do, all's well.